spec/unit/action/runner_spec.rb

```diff
@@ -170,6 +171,37 @@ def call
     end
   end
 
+  describe "configuring git credentials" do
+    it 'errors if both token and key paths are passed' do
+      runner = described_class.new({ 'token-path': '/nonexistent',
+                                     'sshkey-path': '/also/fake' }, %w[args yes], action_class)
+      expect{ runner.call }.to raise_error(R10K::Error, /Cannot specify both/)
+    end
+
+    it 'passes sshkey path to deployment' do
+      sshkey_file = Tempfile.new('sshkey')
+      runner = described_class.new({ 'sshkey-path': sshkey_file.path }, %w[args yes], action_class)
+      runner.call
+      expect(runner.instance.settings[:git][:private_key_from_cli]).to eq(sshkey_file.path)
+    end
+
+    it 'saves the token path in settings hash' do
+      token_file = Tempfile.new('token')
+      token_file.write('my_token')
+      token_file.close
+      runner = described_class.new({ 'token-path': token_file.path }, %w[args yes], action_class)
+      runner.call
+      expect(runner.instance.settings[:git][:oauth_token_file]).to eq(token_file.path)
+    end
+
+    it 'saves the token from stdin' do
+      allow($stdin).to receive(:read).and_return("my_token")
```

Comment:
I guess this isn't needed anymore, since we do the reading in the credentials class instead.